### DISCUSSION TITLE
Build hangs on running "flutter build ipa" with --export-options-plist

### DISCUSSION THREAD

**Discussion by ebragaparah** [codemagic-ci-cd/codemagic-docs#2148] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I am using Flutter 3.13.8, with Xcode 14.2 and it is a M1 Mac. everything builds fine on local but when I run the pipeline, it just freezes.
CocoaPods is also updated, version 1.13.0, on CodeMagic and local.

### Expected results

Build pass on Codemagic like it does on local.

### Actual results

Build hangs on running build command.

```
[   +2 ms] Running Xcode build...
[        ] executing: [/Users/builder/clone/ios/] /usr/bin/arch -arm64e xcrun xcodebuild -configuration Release VERBOSE_SCRIPT_LOGGING=YES -workspace Runner.xcworkspace -scheme Runner -sdk iphoneos -destination generic/platform=iOS -resultBundlePath /var/folders/w2/rrf5p87d1bbfyphxc7jdnyvh0000gn/T/flutter_tools.X1Rs7w/flutter_ios_build_temp_dirB4wCvM/temporary_xcresult_bundle -resultBundleVersion 3 FLUTTER_SUPPRESS_ANALYTICS=true COMPILER_INDEX_STORE_ENABLE=NO -archivePath /Users/builder/clone/build/ios/archive/Runner archive
```

### Build id (optional)

65386608aa5a1171adc8536f

**Comment by himesh-cm** [comment]
Hi @ebragaparah, can you run a build without `--verbose` for the flutter build ipa command and share a new build ID?

**Reply by ebragaparah** [reply]
Sure, @himesh-cm! Thanks
id: 653ac538146a4830e865cd9c

**Reply by ebragaparah** [reply]
Hi, @himesh-cm. As you can see, the build just works without the `--vebose` flag.

But I noticed a thing: 
Yesterday I tried to build also with Xcode 15.1 on Codemagic and I didn't get stuck, I received
```
[  +13 ms] Failed to build iOS app
[ +107 ms] Error (Xcode): Cycle inside <TARGET>; building could produce unreliable results.
           Cycle details:
           → Target <TARGET>: CodeSign /Users/builder/Library/Developer/Xcode/DerivedData/.../<TARGET>.app
           ○ That command depends on command in Target <TARGET>: script phase “[CP] Embed Pods Frameworks”
           ○ Target <TARGET> has copy command from '/Users/builder/Library/Developer/Xcode/DerivedData/.../OneSignalNotificationServiceExtension.appex'
           ○ That command depends on command in Target <TARGET>: script phase “Thin Binary”
           ○ Target <TARGET> has process command with output '/Users/builder/Library/Developer/Xcode/DerivedData/.../<TARGET>.app/Info.plist'
           ○ Target <TARGET> has copy command from '/Users/builder/Library/Developer/Xcode/DerivedData/.../OneSignalNotificationServiceExtension.appex'
```
instead.

So I changed the order of steps on Build Phases. I don't know if it can have any influence.
Thanks

**Reply by himesh-cm** [reply, marked as answer]
@ebragaparah, 

The timeout may be due to verbose builds, which take time to print logs to the UI. Verbose builds should only be used for debugging, not regular workflows. Please avoid using verbose builds.

Is there anything else I can assist you with?
